corehq/apps/registration/utils.py

```diff
@@ -131,17 +132,39 @@ def request_new_domain(request, form, is_new_user=True):
 
     if is_new_user:
         dom_req.save()
-        send_domain_registration_email(request.user.email,
-                                       dom_req.domain,
-                                       dom_req.activation_guid,
-                                       request.user.get_full_name(),
-                                       request.user.first_name)
+        if settings.IS_SAAS_ENVIRONMENT:
+            load_appcues_template_apps.apply_async(
+                (new_domain.name, current_user.username),
+                link=send_domain_registration_email(
```

Comment:
this is pretty cool. I don't think we've used this pattern anywhere else on HQ. Are you sure that this works as expected? I think this just calls the `send_domain_registration_email` function when evaluating the `apply_async` kwargs (i.e. it sends the email before actually registering the celery task). The docs suggest that you should pass a celery `signature` here: http://docs.celeryproject.org/en/latest/reference/celery.html#celery.Signature